draft-ietf-taps-interface.md

```diff
@@ -775,9 +772,10 @@ client-server interactions. Active open is supported by this interface through t
 Initiate Action:
 
 ~~~
-Connection := Preconnection.Initiate()
+Connection := Preconnection.Initiate(timeout)
```

Comment:
Why do we need a timeout as a mandatory parameter here? I am fine with adding an option one, as we have no other means for this yet, but mandatory?

Reply:
I'd say this is a Connection Property rather than a Selection Property, so it should go in Section 9.1 -- in fact it already is there: 9.1.4 is "Timeout for aborting Connection" (which is missing a Type, but we can easily add that).

So let's not add it as a mandatory parameter here, but have it as a Connection Property with some default.

Reply:
I didn't mean to make it mandatory, sorry - happy to make it optional. Having it here seems better: all the other properties express a wish (or requirement, or wish to avoid, ... etc.) about the Connection that we're about to get - but this one doesn't. It really stands out in the list, as being nothing but a detail of how init works - so I thought it seems much cleaner to have it here as a parameter.

Reply:
Wait, Wishes (or requirements etc) are Selection Properties, not Connection properties. 
Connection properties do not express a wish (or requirement etc), but they set a specific variable, such as a timeout, to a specific value, such as a number of seconds.

I see your point that it's not really the property of a Connection, but of a connection attempt or something like this.
But then, does the Connection property in 9.1.4 make sense? Or is this yet something else?

Reply:
It's something else - but now I notice that the wording there ("before deciding that a Connection has failed after establishment.") is a bit misleading. This (9.1.4) is the timeout that decides to abort a connection when data could not be delivered too long. It can be changed at any time, whereas the init. timeout just decides how long to try establishing before giving up.

Reply:
.... and now I noticed how my answer to you was confusing. Sorry - indeed, you talked about the Connection Property which is already there. This update (of making the timeout a (with the next commit, optional) parameter of init) is about removing the init timeout only, 5.2.14.

Reply:
Okay, makes sense :)
Then I'm fine with this timeout being an optional parameter to Initiate() (Maybe it's better to say explicitly that it's optional?), but I'd like it if 9.1.4 was made a bit more clear.

Reply:
In general, I think we need to handle optional parameters better in the API doc. Will file an issue.

Reply:
done - I added "optional: " in front of the parameter. I suggest to accept this as-is for now and fix afterwards because it's a general style thing - if you don't like this specific style proposal, no problem to change it.

Reply:
Changed to @britram style optional in the last commit.